### PR TITLE
chore(experiments): Rename experimentData to experiment everywhere

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

`experimentData` is not used anymore, since we've moved to a kea-form where `experiment` is the new `experimentData`

## Changes

see^

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/experiments/experimentLogic.tsx [hunk 1]
```diff
@@ -524,58 +519,54 @@ export const experimentLogic = kea<experimentLogicType>([
...
                 },
             ],
         ],
         variants: [
-            (s) => [s.experiment, s.experimentData],
-            (newExperimentData, experimentData): MultivariateFlagVariant[] => {
-                if (experimentData?.start_date) {
-                    return experimentData?.parameters?.feature_flag_variants || []
+            (s) => [s.experiment, s.experiment],
+            (newexperiment, experiment): MultivariateFlagVariant[] => {
```
Comment (liyiy) [inline frontend/src/scenes/experiments/experimentLogic.tsx:537]: `newExperiment` and similarly below too

**Comment by liyiy** [comment]
I know this is a simple PR but can we write a 1 line explanation on why we're doing this in the PR summary 😁

File: frontend/src/scenes/experiments/experimentLogic.tsx (hunk 1 above)
Comment (neilkakkar) [reply frontend/src/scenes/experiments/experimentLogic.tsx:537]: D'oh, accidentally added this to the follow up instead: https://github.com/PostHog/posthog/pull/13343

**Comment by neilkakkar** [comment]
It's a leftover from refactoring, have updated description

**Pull request merged**